File: java/src/main/java/com/meteroid/models/InvoiceType.java
// This file is @generated
package com.meteroid.models;

import com.fasterxml.jackson.annotation.JsonValue;
import com.meteroid.Utils.ToQueryParam;

public enum InvoiceType implements ToQueryParam {
    RECURRING("RECURRING"),
    ONE_OFF("ONE_OFF"),
    ADJUSTMENT("ADJUSTMENT"),
    USAGE_THRESHOLD("USAGE_THRESHOLD");
    private final String value;

    InvoiceType(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }

    @Override
    public String toQueryParam() {
        return this.value;
    }
}

File: python/meteroid/models/invoice_type.py
# this file is @generated
import enum


class InvoiceType(str, enum.Enum):
    RECURRING = "RECURRING"
    ONE_OFF = "ONE_OFF"
    ADJUSTMENT = "ADJUSTMENT"
    USAGE_THRESHOLD = "USAGE_THRESHOLD"

    def __str__(self) -> str:
        return str(self.value)
